### ISSUE TITLE
settings.py problem

### ISSUE THREAD

**Issue by bhbraswell** [Applied-GeoSolutions/gips#14]
There still seems to be an issue with recognizing a local settings.py

As a test, I changed the archive location for MERRA in my local copy of settings.py, but it is still looking in the system level archive, apparently. There are no files in this archive, but I get the following result.

```
(venv)(screen)rbraswell@rio:~/repo/gips$ gips_inventory merra -p prcp tave -d 2000-001 -t h01v01 -v4
GIPS Data Inventory (v0.8.1)
Retrieving inventory for site tiles

Asset Holdings
    DATE     PRECTOT   PROFILE   PROFILEP    T10M      T2M        TS     Product  
2000        
    001       100.0%                                   100.0%               PRECTOT  T2M  tave  tmax  tmin


1 files on 1 dates

SENSORS

```

**Comment by ircwaves** [comment]
Are you in a virtual environment inside another virtual environment? (venv)(screen)  Or does usage of `screen` just prepend the prompt as well?

All that aside, as shown below, I cannot reproduce this...unless you've already fixed it.
```
icooke@rio:~$ cd /home/rbraswell/repo/gips
icooke@rio:/home/rbraswell/repo/gips$ . venv/bin/activate
(venv)icooke@rio:/home/rbraswell/repo/gips$ cd
(venv)icooke@rio:~$ gips_config print
GIPS Configuration Utility (v0.8.1)

DATABASES
{'geokit': {'HOST': 'indus',
            'NAME': 'geokit',
            'PASSWORD': 'geokitp4ss',
            'PORT': '5432',
            'USER': 'geokit'}}

EMAIL
'rbraswell@appliedgeosolutions.com'

REPOS
{'aod': {'repository': '/titan/data/aod'},
 'landsat': {'6S': False,
             'MODTRAN': False,
             'extract': False,
             'repository': '/titan/data/landsat'},
 'merra': {'repository': '/titan/data/merra2/merra'},
 'modis': {'repository': '/titan/data/modis'},
 'weld': {'repository': '/titan/data/weld'}}
(venv)icooke@rio:~$ gips_inventory merra -p prcp tave -d 2000-001 -t h01v01 -v4
/usr/lib/python2.7/dist-packages/pkg_resources.py:1031: UserWarning: /home/icooke/.python-eggs is writable by group/others and vulnerable to attack when used with get_resource_filename. Consider a more secure location (set with .set_extraction_path or the PYTHON_EGG_CACHE environment variable).
  warnings.warn(msg, UserWarning)
GIPS Data Inventory (v0.8.1)
Retrieving inventory for site tiles

Asset Holdings
No matching files in inventory
```

**Comment by bhbraswell** [comment]
Weird. Ok thanks I'll check. Screen is just screen, no env of its own as far as I know.

Rob

**Comment by bhbraswell** [comment]
I can't reproduce your not being able to reproduce. I mean: I logged out and back in again, and then tried the same, with no change in result. So it matters that I'm me and not you, but why?

```
rbraswell@rio:~$ cd repo/gips
rbraswell@rio:~/repo/gips$ source venv/bin/activate
(venv)rbraswell@rio:~/repo/gips$ gips_config print
GIPS Configuration Utility (v0.8.1)

DATABASES
{'geokit': {'HOST': 'indus',
            'NAME': 'geokit',
            'PASSWORD': 'geokitp4ss',
            'PORT': '5432',
            'USER': 'geokit'}}

EMAIL
'admin@appliedgeosolutions.com'

REPOS
{'aod': {'repository': '/titan/data/aod'},
 'daymet': {'repository': '/titan/data/daymet'},
 'landsat': {'6S': True,
             'MODTRAN': True,
             'extract': False,
             'repository': '/titan/data/landsat'},
 'merra': {'repository': '/titan/data/merra'},
 'modis': {'repository': '/titan/data/modis'},
 'weld': {'repository': '/titan/data/weld'}}

```

**Comment by ircwaves** [comment]
The issue is the existence `/home/rbraswell/.gips/settings.py` AND that there is no indication that the settings are pulled from that file.

I have fallen to this one before as well.  Perhaps a good practice would be to  toss a pointless entry into the REPOS['driver'] dictionary to remind ourselves like this:

```
gips_config print
   [SNIP]
 'modis': {'repository': '/titan/data/modis'
           'from_user_settings': True},
  [SNIP]
```

**Comment by bhbraswell** [comment]
Thanks. Is this a bug, or like a bug? I think the use of hidden .gips in this case for settings is a little obscure and undesirable. But in any case if there is more than one way to have custom settings (hidden, and in the gips/gips/ folder) then that's probably not good.

**Comment by bhbraswell** [comment]
It also seems to not recognize either the tiles.shp file in the driver, or the file pointed to by the 'tiles' keyword in settings, which is maybe its own problem, or maybe related.

**Comment by bhbraswell** [comment]
If I use ~/.gips/settings.py then it will recognize the tiles.shp location.

**Comment by ircwaves** [comment]
In this case, I think it is a misconf due to missing documentation.

I think that the trick is that `.gips/settings.py` is really only for the external gips-compliant driver case.  I think the ideal resolution would be if user settings were automagically augmented with an attribute alluding to their origin.  I would think making the user settings (and system settings) file a JSON file or something that `gips.settings.py` loads and processes would be a safer way to go than just importing a `~/.gips/settings.py` file.

**Comment by demiurg** [comment]
I just chatted with Rob and I think it's a good idea to have single cascading settings import in order: system < user home < local dir
whereby local is last and overrides things. Right now, it seems that home overrides local.

**Comment by ircwaves** [comment]
Per our conversation, it seems like there is an aspect of settings that bumps up against ideas of having the assets and products be more secure, and we should consider this for future work.  In the interim, it seems like there isn't really a use-case for `~/.gips/settings.py` other than top secret stuff (external gips-compliant drivers), and that until we have another use-case, the move is to edit the moduel settings file that is created via `gips_config env`.

**Comment by ircwaves** [comment]
Pavel and I just chatted more, and he is going to work on some backwards compatible settings changes.

I am going to test out using `.gips/settings.py` to enable development of a prism data plugin/driver.  Then once the driver is done, I will see what issues arise in moving it into gips proper.

**Comment by ags-tolson** [comment]
Closing as invalid because:
* No consensus on what the real problem was
* No clear idea of what changes or work are required
* No code anywhere

Instead of re-opening, suggest first getting some clarity on what work is required, then making a new issue and referencing this one.

**Comment by ags-tolson** [comment]
PS - also closing because it sat idle for 1.5 years
